File: app/Models/PaymentMethod.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

#[Fillable(['brand_id', 'name', 'code', 'sort_order', 'is_active', 'requires_cash_input', 'qrcode_image', 'account_name', 'account_number'])]
class PaymentMethod extends Model
{
    public function brand(): BelongsTo
    {
        return $this->belongsTo(Brand::class);
    }

    protected function casts(): array
    {
        return [
            'is_active'           => 'boolean',
            'requires_cash_input' => 'boolean',
        ];
    }
}

File: mobile/lib/models/payment_method.dart
class PaymentMethod {
  final int id;
  final String name;
  final String code;
  final bool requiresCashInput;
  final String? qrcodeImage;

  PaymentMethod({
    required this.id,
    required this.name,
    required this.code,
    required this.requiresCashInput,
    this.qrcodeImage,
  });

  factory PaymentMethod.fromJson(Map<String, dynamic> json) {
    return PaymentMethod(
      id: json['id'] as int,
      name: json['name'] as String,
      code: json['code'] as String,
      requiresCashInput: json['requires_cash_input'] as bool? ?? false,
      qrcodeImage: json['qrcode_image'] as String?,
    );
  }
}
